packages/smart-wallet/src/smartWallet.js

```diff
@@ -745,6 +757,38 @@ export const prepareSmartWallet = (baggage, shared) => {
           }
           return baggage.get(state.address);
         },
+        /** @param {string} suggestion */
+        findUnusedName(suggestion) {
+          const { myStore } = this.state;
+          let nonce = 0;
+          let name = suggestion;
+          while (myStore.has(name)) {
+            nonce += myStore.getSize(); // avoid linear work
+            name = `${suggestion}.${nonce}`;
+          }
+          return name;
+        },
+        /**
+         * @param {ResultPlan} plan
+         * @param {unknown} result
+         */
+        saveOfferResult(plan, result) {
+          const { myStore } = this.state;
+          const name = plan.overwrite
+            ? plan.name
+            : this.facets.helper.findUnusedName(plan.name);
+
+          if (myStore.has(name)) {
+            if (plan.overwrite) {
```

Comment:
I like the `const name` and ending with `return name;` but the two checks for `plan.overwrite` mean if `findUnusedName` returns a dupe that the value will get lost.

Please include tests for the matrix of overwrite T/F and present T/F.